src/psnawp_api/utils/misc.py

```diff
@@ -1,8 +1,30 @@
 from __future__ import annotations
 
+import base64
+import binascii
+import re
 from datetime import datetime
 from typing import Optional
 
+DECODED_NPID_PATTERN = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
```

Comment:
since the last two characters of the string are going to be ISO 3166-1 alpha-2 you might not need regex. 

Compare the regex to simply slicing last two characters. I think slicing will be faster.

Reply:
https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2